test/io/database_parameters_test.cpp

```diff
@@ -120,3 +120,32 @@ TEST_F(Database_parameter_test, matrix3x3_parameter)
         for (size_t j = 0; j < 3; ++j)
             EXPECT_DOUBLE_EQ(s.m(i,j), m_expected(i,j));
 }
+
+TEST_F(Database_parameter_test, unused_parameter)
+{
+    class Sample_class
+    {
+     public:
+        double d1;
+        double d2;
+        DECLARE_PARAMS({"double_parameter",d1},{"double_parameter_2",d2});
```

Comment:
Add a second test where you only declare "double_parameter_2" and you check that "double_parameter" is not used